src/model/CoilSystemCoolingWater.cpp

```diff
@@ -0,0 +1,452 @@
+      const bool result = setPointer(OS_CoilSystem_Cooling_WaterFields::CompanionCoilUsedForHeatRecovery, companionCoilUsedForHeatRecovery.handle());
+      return result;
+    }
+
+    void CoilSystemCoolingWater_Impl::resetCompanionCoilUsedForHeatRecovery() {
+      const bool result = setString(OS_CoilSystem_Cooling_WaterFields::CompanionCoilUsedForHeatRecovery, "");
+      OS_ASSERT(result);
+    }
+
+    boost::optional<Schedule> CoilSystemCoolingWater_Impl::optionalAvailabilitySchedule() const {
+      return getObject<ModelObject>().getModelObjectTarget<Schedule>(OS_CoilSystem_Cooling_WaterFields::AvailabilityScheduleName);
+    }
+
+    boost::optional<WaterToAirComponent> CoilSystemCoolingWater_Impl::optionalCoolingCoil() const {
+      return getObject<ModelObject>().getModelObjectTarget<WaterToAirComponent>(OS_CoilSystem_Cooling_WaterFields::CoolingCoil);
+    }
+
+    boost::optional<WaterToAirComponent> CoilSystemCoolingWater_Impl::optionalCompanionCoilUsedForHeatRecovery() const {
+      return getObject<ModelObject>().getModelObjectTarget<WaterToAirComponent>(OS_CoilSystem_Cooling_WaterFields::CompanionCoilUsedForHeatRecovery);
+    }
+
+  }  // namespace detail
+
+  CoilSystemCoolingWater::CoilSystemCoolingWater(const Model& model) : StraightComponent(CoilSystemCoolingWater::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::CoilSystemCoolingWater_Impl>());
+
+    bool ok = true;
+    auto alwaysOn = model.alwaysOnDiscreteSchedule();
+    ok = setAvailabilitySchedule(alwaysOn);
+    OS_ASSERT(ok);
+
+    CoilCoolingWater coolingCoil(model);
+    ok = setCoolingCoil(coolingCoil);
+    OS_ASSERT(ok);
+    ok = setDehumidificationControlType("None");
+    OS_ASSERT(ok);
+    ok = setRunonSensibleLoad(true);
+    OS_ASSERT(ok);
+    ok = setRunonLatentLoad(false);
+    OS_ASSERT(ok);
+    ok = setMinimumAirToWaterTemperatureOffset(0.0);
+    OS_ASSERT(ok);
+    ok = setEconomizerLockout(true);
+    OS_ASSERT(ok);
+    ok = setMinimumWaterLoopTemperatureForHeatRecovery(0.0);
+    OS_ASSERT(ok);
+  }
+
+  CoilSystemCoolingWater::CoilSystemCoolingWater(const Model& model, const WaterToAirComponent& coolingCoil)
+    : StraightComponent(CoilSystemCoolingWater::iddObjectType(), model) {
+    OS_ASSERT(getImpl<detail::CoilSystemCoolingWater_Impl>());
+
+    bool ok = true;
+    auto alwaysOn = model.alwaysOnDiscreteSchedule();
+    ok = setAvailabilitySchedule(alwaysOn);
+    OS_ASSERT(ok);
+
+    ok = setCoolingCoil(coolingCoil);
+    OS_ASSERT(ok);
+    ok = setDehumidificationControlType("None");
+    OS_ASSERT(ok);
+    ok = setRunonSensibleLoad(true);
+    OS_ASSERT(ok);
+    ok = setRunonLatentLoad(false);
+    OS_ASSERT(ok);
+    ok = setMinimumAirToWaterTemperatureOffset(0.0);
+    OS_ASSERT(ok);
+    ok = setEconomizerLockout(true);
+    OS_ASSERT(ok);
+    ok = setMinimumWaterLoopTemperatureForHeatRecovery(0.0);
+    OS_ASSERT(ok);
```

Comment:
All good too, we could have used a helper method assignEnergyPlusIDDDefaults() or something.

Reply:
868633ab378788f7205c73dda7d0432d5c593841